### DISCUSSION TITLE
Sample of web frontend to imodel backend opening snapshot

### DISCUSSION THREAD

**Discussion by MattPil29** [iTwin/community#52] [Q&A]
Hi,

I'm looking for sample or at least some guidance/code snippets to pull together the following:
1) standard react web front end (covered [here](https://www.itwinjs.org/learning/tutorials/develop-web-viewer/)) with the code to connect to our own node.js backend - how do we do this without an app id etc which all the samples promote?
2) quick sample of spinning up a node.js backend to read a asnapshot/local BIM file.  I see the agent turoial [here](https://www.itwinjs.org/learning/tutorials/develop-agent/) but again it needs ids and not file names.

I'm assuming somewhere in the agent sample I'd need to swap in some code to use https://www.itwinjs.org/reference/core-backend/imodels/snapshotdb/#openfile

Thanks in advance for any help given, regards, Matt

**Comment by hl662** [comment]
Hi Matt, see this [sample](https://github.com/hl662/sample-node-imodel) I spun up for opening a node backend to read a BIM file. You can run 'pnpm i', and then 'pnpm start' (I used pnpm for my package manager, but npm should be okay too?)

As for your first point, can I have a clearer idea of what you are trying to do? A node backend that opens imodels locally, and your frontend app can interact with it?
